llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -194,6 +194,10 @@ class LocationSize {
     return hasValue() && getValue() == Other;
   }
 
+  bool operator<=(const LocationSize &Other) const {
+    return Value <= Other.Value;
```

Comment:
LocationSize has a whole bunch of special values and bits that make this incorrect. You need to work on getValue() for the cases where it is valid.

Reply:
Not really sure why I didn't use getValue in the first place.